aten/src/ATen/native/TensorShape.cpp

```diff
@@ -41,8 +41,14 @@ std::vector<Tensor> chunk(const Tensor& self, int64_t chunks, int64_t dim) {
     AT_ERROR("chunk expects `chunks` to be greater than 0, got: ", chunks);
   }
   int64_t split_size = (self.size(dim) + chunks - 1) / chunks;
-  // ensure this is dispatched through Tensor/Type, rather than the native function directly.
-  return self.split(split_size, dim);
+
+  // We need to call split_with_sizes in the case where split_size and dimension size are 0, because
+  // a call to split would discard the number of chunks (because we can have an arbitrary number of
+  // 0-sized chunks adding up to 0).  So, call split_with_sizes with the correct number of chunks,
+  // and do this for all cases for consistency.
+  std::vector<int64_t> split_sizes(chunks, split_size);
```

Comment:
I guess this is fine, because it doesn't change the asymptotics at all.

Reply:
Right.  Honestly, split should probably just call split_sizes as well because it's better to just have one implementation than worry about something that isn't going to affect the performance.